src/main/docker/startup/startup.sh

```diff
@@ -17,15 +17,15 @@
 
 # Create a convenience function for logging
 log() {
-    echo "${0##*/}: $@"
+    echo "[$(date +%H:%M:%S.%3NZ) #$(printf '%03X\n' $$).??? INFO -            -   ] ${0##*/}: $@"
 }
 
 # Run the executable scripts that are in the drop-in folder
 log "Running startup scripts..."
 for script in $(dirname "$0")/startup.d/*; do
     if [ -x "$script" ]; then
         log "Running ${script##*/}..."
-        "$script" |& sed -ue "s/^/${script##*/}: /"
+        "$script" |& sed -ure "s/^warning:/WARN:/I; /^(info|error|warn|debug|trace):/I!s/^/info: /; s/^([^:]*): ?(.*)$/[$(date +%H:%M:%S.%3NZ) #$(printf '%03X\n' $BASHPID).??? \U\1\E -            -   ] ${script##*/}: \2/"
```

Comment:
1. The time being reported would appear to be the time when the script first started running rather than the time that it logged the statement.
2. The log level doesn't seem to be aligned with the caf-logging format (i.e. it doesn't have the width set to 5).

Reply:
Fixed 1. Working on 2.

Reply:
For item no1 : 
The string format to fix string width example: printf ':%-5.5s:' 1234 output - :1234 :
When the same pattern applied for backreference \1, it is not evaluated properly.
**Approach 1 :** I tried with variable substitution also inside printf like 
log='\1'
$(printf '%-5.5s' $log) it is giving below unexpected output

[12:28:26.151Z #5B43.??? warn   -            -   ] startup1.sh: This is test warn from STATUP 1
[12:28:26.151Z #5B43.??? WARN   -            -   ] startup1.sh: This is test WARNING from STATUP 1
[12:28:26.151Z #5B43.??? error   -            -   ] startup1.sh: 1.This is test error from Startup 1
**Approach 2:**
$(printf '%-5.5s' $(printf '%s' '\1'))
output:
[12:34:13.049Z #5FBA.??? warn   -            -   ] startup1.sh: This is test warn from STATUP 1
[12:34:13.049Z #5FBA.??? WARN   -            -   ] startup1.sh: This is test WARNING from STATUP 1
[12:34:13.049Z #5FBA.??? error   -            -   ] startup1.sh: 1.This is test error from Startup 1

I am trying to fix the printf evaluation inside printf to substitute the actual loglevel value for formatting.
The formatting working fine for plain string

Reply:
This is the approach you tried to default the log level to INFO when no log level was available. 
As we discovered then it is not possible to do this with a back reference.

Reply:
Since formatting is not working for backreference, I added one more sed expression to pad one space to log level with 4 characters. Here only **info** and **warn** has 4 characters. These 2 will be padded with one space to right and before : 
s/^(\w{0,4}):/\1 :/I;
Then This will enter to final expression for the actual replacement

Reply:
> Fixed 1. Working on 2.

1. is the time issue.  I've no idea why you think it's fixed because I've just tested it and it still doesn't work.